### PR TITLE
Create docker push workflow

### PR DESCRIPTION
**Pull request by indykoning**
This will automatically push docker images into the github container repository under the organisation

### CODE REVIEW

**Comment by zachgoll** [comment]
Hey @indykoning this is great, just want to make sure I'm following—this setup will push a _future_ `Dockerfile` (that hasn't been created yet) to Github Package repository correct?

File: .github/workflows/push-docker-container.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: 'Additional tag to push'
+        required: false
+  push:
+    branches:
+      - 'main'
+    tags:
+      - '*'
```
Comment (tebayoso) [inline .github/workflows/push-docker-container.yml:13]: I would recommend to push only certain tags, for example:
```suggestion
      - 'RC*'
      - 'preview*
```

Comment (indykoning) [reply .github/workflows/push-docker-container.yml:13]: During development i absolutely agree, though wouldn't we want real version numbers such as `1.0.0` or `2024.2` to have their image created as well?

**Comment by zachgoll** [comment]
@indykoning this is great, but our Dockerfile is not quite ready and we'll want to make sure that once we start pushing images to GHCR, all files work.

I think we can either close this out and come back to it later or make the `Dockerfile` updates directly in this PR

**Comment by zachgoll** [comment]
Closing out, will reference this when we introduce official support for a Dockerfile.

**Pull request closed without merging**